258 replace mock listings in my listings page

JorgeRivera94 opened the pull request:
Closes #258 

* Updated listings Collection to include a “user” field. In the template, the type for “user” is set to String. The “user” field should be a Reference to a users document. It was set to String because the template does not allow setting it to reference type without linking to an existing document.
* Created a users document with my credentials through the Sign Up page.
  * id: o7O7aJAdDsgLMeNjeP9U87Jj4ob2
  * createdAt: 9 de abril de 2025, 1:21:49 p.m. UTC-4
  * email: "jorge.rivera94@upr.edu"
  * name: "Jorge Rivera"
  * username: "KFall"
* Edited API to read from the current user
  * mock_backend/mockApi.ts
  * To preserve familiarity and compatibility, mostly preserved function names and uses.
  * Added functions needed to interpret the listing values.
* Edited DisplayMyListing() to interact with the edited mock API.
* Created a mock listings document
  * listings/Mock_My_Listings_Book
* Created a mock listing_image
  * listing_images/Mock_My_Listings_Image
* Debugged with try-catch blocks and ensured the displayed listing is read from firebase data.

JuanQ1226 reviewed (commented):
This replaces the mock api for the actual db?

JorgeRivera94 commented:
Replaces the mock API for reading (displaying) the listing. I also changed the other mock API functions for the My Listings page for consistency, so they all interact with firestore instead of the mock json data.

This sets base for the integration of the editing and deleting functionalities of the page, but dedicated work to ensure their functioning and testing are out of scope for this issue. These can be assigned to Listings and Search team developers to ensure their proper functioning.

JuanQ1226 commented:
I think you should do some kind of refactoring because the files are still named mockApi for example, even though the api is actually integrated.

JorgeRivera94 commented:
> I think you should do some kind of refactoring because the files are still named mockApi for example, even though the api is actually integrated.

Changed directory and file names, and removed mocked JSON data. Also updated include paths for the new names.

evnd04 commented:
@JuanQ1226 @JorgeRivera94  is this still a work in progress?

JuanQ1226 commented:
@JorgeRivera94 Is the refactoring done?

JorgeRivera94 commented:
@JuanQ1226 @evnd04 Refactoring is done.

evnd04 commented:
@JuanQ1226 Whenever posible verify this PR to see if it satisfies your requirements

JuanQ1226 reviewed (approved):
LGTM
